// @mui/x-charts/RadarChart/RadarAxisHighlight/radarAxisHighlightClasses.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.chartsAxisHighlightClasses = void 0;
exports.getRadarAxisHighlightUtilityClass = getRadarAxisHighlightUtilityClass;
exports.useUtilityClasses = void 0;
var _composeClasses = _interopRequireDefault(require("@mui/utils/composeClasses"));
var _generateUtilityClass = _interopRequireDefault(require("@mui/utils/generateUtilityClass"));
var _generateUtilityClasses = _interopRequireDefault(require("@mui/utils/generateUtilityClasses"));
/**
 * @deprecated Use `RadarClasses` instead.
 */

/**
 * @deprecated Use `RadarClassKey` instead.
 */

/**
 * @deprecated Use `getRadarUtilityClass` instead.
 */
function getRadarAxisHighlightUtilityClass(slot) {
  return (0, _generateUtilityClass.default)('MuiRadarAxisHighlight', slot);
}

/**
 * @deprecated Use `radarClasses` instead.
 */
const chartsAxisHighlightClasses = exports.chartsAxisHighlightClasses = (0, _generateUtilityClasses.default)('MuiRadarAxisHighlight', ['root', 'line', 'dot']);

/**
 * @deprecated Use `useUtilityClasses` instead.
 */
const useUtilityClasses = classes => {
  const slots = {
    root: ['root'],
    line: ['line'],
    dot: ['dot']
  };
  return (0, _composeClasses.default)(slots, getRadarAxisHighlightUtilityClass, classes);
};
exports.useUtilityClasses = useUtilityClasses;
